src/main/java/uk/gov/hmcts/reform/sscs/service/CcdNotificationsPdfService.java

```diff
@@ -118,6 +111,35 @@ public SscsCaseData mergeLetterCorrespondenceIntoCcd(byte[] pdf, Long ccdCaseId,
         return caseDetails.getData();
     }
 
+    public void mergeLetterCorrespondenceIntoCcdV2(byte[] pdf, Long ccdCaseId, Correspondence correspondence) {
+        mergeLetterCorrespondenceIntoCcdV2(pdf, ccdCaseId, correspondence, DEFAULT_SENDER_TYPE);
+    }
+
+    public void mergeLetterCorrespondenceIntoCcdV2(byte[] pdf, Long ccdCaseId, Correspondence correspondence, String senderType) {
+        final List<Correspondence> correspondences = getCorrespondences(pdf, correspondence);
+
+        Consumer<SscsCaseData> caseDataConsumer = sscsCaseData -> {
+            List<Correspondence> existingCorrespondence = sscsCaseData.getCorrespondence() == null ? new ArrayList<>() : sscsCaseData.getCorrespondence();
+            List<Correspondence> allCorrespondence = new ArrayList<>(existingCorrespondence);
+            allCorrespondence.addAll(correspondences);
+            allCorrespondence.sort(Comparator.reverseOrder());
+            sscsCaseData.setCorrespondence(allCorrespondence);
+        };
+
+        String description = String.format("Notification sent via %s", senderType);
+        try {
+            updateCcdCaseService.updateCaseV2(ccdCaseId,
```

Comment:
can we please add some logs so that it is evident v2 is used ? Easy to debug later ?